### ISSUE TITLE
Completely broken

### ISSUE THREAD

**Issue by gaetgu** [vlang/vorum#24]
It seems that this app was written with an older version of V. The compiler does not accept some older syntax used in the app. Here are my issues running POP!_OS latest stable.

(This one shows up right away. obviously deprecated)
```console
./main.v:3:8: error: `import()` has been deprecated, use `import x` instead
    1 | module main
    2 | 
    3 | import (
      |        ^
    4 |     vweb
    5 |     pg
```

Once I fix that deprecated piece of code, this is printed:

```console
./main.v:20:13: warning: use `(mut f Foo)` instead of `(f mut Foo)`
   18 | }
   19 | 
   20 | pub fn (app mut App) reset() {
      |             ~~~
   21 |     app.user = User{}
   22 |     app.logged_in = false
./main.v:31:13: warning: use `(mut f Foo)` instead of `(f mut Foo)`
   29 | }
   30 | 
   31 | pub fn (app mut App) init() {
      |             ~~~
   32 |     db := pg.connect(pg.Config{host:'127.0.0.1', dbname:db_name, user:db_user}) or { panic(err) }
   33 |     app.db = db
./main.v:37:13: warning: use `(mut f Foo)` instead of `(f mut Foo)`
   35 | }
   36 | 
   37 | pub fn (app mut App) index() {
      |             ~~~
   38 |     app.auth()
   39 |     posts := app.find_all_posts()
./main.v:46:13: warning: use `(mut f Foo)` instead of `(f mut Foo)`
   44 | // TODO ['/post/:id/:title']
   45 | // TODO `fn (app App) post(id int)`
   46 | pub fn (app mut App) post() {
      |             ~~~
   47 |     id := app.get_post_id()
   48 |     post := app.retrieve_post(id) or {
./main.v:59:13: warning: use `(mut f Foo)` instead of `(f mut Foo)`
   57 | 
   58 | // new post
   59 | pub fn (app mut App) new() {
      |             ~~~
   60 |     app.auth()
   61 |     $vweb.html()
./main.v:65:13: warning: use `(mut f Foo)` instead of `(f mut Foo)`
   63 | 
   64 | // [post]
   65 | pub fn (app mut App) new_post() {
      |             ~~~
   66 |     app.auth()
   67 |     mut name := ''
./main.v:86:9: warning: use `(mut f Foo)` instead of `(f mut Foo)`
   84 | 
   85 | // [post]
   86 | fn (app mut App) comment() {
      |         ~~~
   87 |     app.auth()
   88 |     post_id := app.get_post_id()
./main.v:106:13: warning: use `(mut f Foo)` instead of `(f mut Foo)`
  104 | }
  105 | 
  106 | pub fn (app mut App) deletepost() {
      |             ~~~
  107 |     app.auth()
  108 |     if !app.user.is_admin {
./main.v:115:17: error: unexpected name `set`
  113 |     db := app.db
  114 |     //db.update Post set nr_comments=10//  is_deleted = true where id = post_id
  115 |     db.update Post set is_deleted = true where id == post_id
      |                    ~~~
  116 |     println('deleted post $post_id')
  117 | }
```

Most of these are just warnings, except that last one (note that line numbers are likely to be different since I fixed the import deprecation):
```console
./main.v:115:17: error: unexpected name `set`
  113 |     db := app.db
  114 |     //db.update Post set nr_comments=10//  is_deleted = true where id = post_id
  115 |     db.update Post set is_deleted = true where id == post_id
      |                    ~~~
  116 |     println('deleted post $post_id')
  117 | }
```

It would be great if you could update this, thanks *so* much.

**Comment by gnanram** [comment]
It would be beneficial for users who want to use/learn Vlang for web development from existing examples. 
could the maintainer of this repo address the compilation issue? even after I have taken care of the reported errors, it is erring out with the below output. 

```
(base) ➜  vorum git:(master)  v .
V panic: array.get: index out of range (i == 1, a.len == 1)
v hash: 758d1f4
0   v                                   0x0000000109566449 array_get + 249
1   v                                   0x00000001097735b4 v__parser__Parser_compile_template_file + 2692
2   v                                   0x0000000109771b7e v__parser__Parser_comptime_call + 10094
3   v                                   0x000000010979aef1 v__parser__Parser_stmt + 3297
4   v                                   0x000000010978daed v__parser__Parser_parse_block_no_scope + 189
5   v                                   0x0000000109794547 v__parser__Parser_fn_decl + 18727
6   v                                   0x00000001097a5445 v__parser__Parser_top_stmt + 165
7   v                                   0x00000001097a0d2c v__parser__Parser_parse + 604
8   v                                   0x00000001097a26a3 v__parser__parse_file + 1555
9   v                                   0x00000001097a6251 v__parser__parse_files + 593
10  v                                   0x00000001097bb9cd v__builder__Builder_front_stages + 205
11  v                                   0x00000001097becca v__builder__Builder_front_and_middle_stages + 74
12  v                                   0x00000001097d6562 v__builder__cbuilder__gen_c + 66
13  v                                   0x00000001097d63e2 v__builder__cbuilder__build_c + 338
14  v                                   0x00000001097d627e v__builder__cbuilder__compile_c + 526
15  v                                   0x00000001097cfd9d v__builder__compile + 413
16  v                                   0x00000001097d7b7a main__main + 5306
17  v                                   0x00000001097db0b7 main + 39
18  v                                   0x0000000109562384 start + 52
19  ???                                 0x0000000000000002 0x0 + 2
(base) ➜  vorum git:(master) 
```
Any help is highly appreciated.

**Comment by ghost** [comment]
Fixed in #27
